bindings/mono/Bindings.cs

```diff
@@ -26,7 +26,19 @@ sealed class Bindings {
 #if UNIX
 		private const string librhash = "librhash.so.0";
 #else
+  #if DARWIN
```

Comment:
Actually compiled C# binary must be platform-independent, so you must detect OS and lib_name at runtime.

Also in cases like this you should use [#elif](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-elif) directive for more clear code.

Reply:
Okay.  Right now, I'm working on a something a little more clear, having the config script actually tell the Makefile through the config.mak what the .DLL is since we know it there.

Reply:
It's good idea for platform-dependent bindings (with compiled C code), like Java/Perl/Ruby, but will not work for Mono and Python.

Suppose you compiled Mono bindings on MSYS with msys-rhash.dll, but library user will run the compiled bytecode on Windows or Linux and the program will fail because of searching Librhash by wrong name.

Reply:
This is one reason why I don't like what I have presented and I'm saying just as much.  BTW, the Python binding seems to be something that just runs in an interpreter and I just put in values for the platform.  I haven't figured out how to do this with NET and C sharp.  From Java, it seems to work with a platform dependent JNI and platform independent classes but I don't know how that comes together.  I have some difficulty getting it work with the Java JDK 9 and from MSYS targeting MINGW-W64.

Reply:
For .Net I've googled the code:
`bool isWindows = System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(OSPlatform.Windows)`

OS text description:
`var osNameAndVersion = System.Runtime.InteropServices.RuntimeInformation.OSDescription`

Also you can check for "MSYS", "MINGW64", "MINGW32" the value of
`Environment.GetEnvironmentVariable("MSYSTEM");`

You still should investigate how to detect cygwin.